From 8b56a0d58905ccdfa27e6562bbe7b0e9e741e9c0 Mon Sep 17 00:00:00 2001
From: "Mazlan, Hazwan Arif" <hazwan.arif.mazlan@intel.com>
Date: Thu, 21 Apr 2022 08:59:59 +0800
Subject: [PATCH] Revert "iris: Return non-zero stride for
 clear color plane"

HSDES ID: 15011043486

This reverts commit db475c81b7cdde9d967b0216e6f12d17e0adae83.
This is workaround solution for Weston issue in ADL family platform.
Please do not upstream this
---
 src/gallium/drivers/iris/iris_resource.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/src/gallium/drivers/iris/iris_resource.c b/src/gallium/drivers/iris/iris_resource.c
index 9f91821d40ef..607b79e816e4 100644
--- a/src/gallium/drivers/iris/iris_resource.c
+++ b/src/gallium/drivers/iris/iris_resource.c
@@ -1658,15 +1658,7 @@ iris_resource_get_param(struct pipe_screen *pscreen,
       }
       return true;
    case PIPE_RESOURCE_PARAM_STRIDE:
-      *value = wants_cc ? 1 :
-               wants_aux ? res->aux.surf.row_pitch_B : res->surf.row_pitch_B;
-
-      /* Mesa's implementation of eglCreateImage rejects strides of zero (see
-       * dri2_check_dma_buf_attribs). Ensure we return a non-zero stride as
-       * this value may be queried from GBM and passed into EGL.
-       */
-      assert(*value);
-
+      *value = wants_aux ? res->aux.surf.row_pitch_B : res->surf.row_pitch_B;
       return true;
    case PIPE_RESOURCE_PARAM_OFFSET:
       *value = wants_cc ? res->aux.clear_color_offset :
-- 
2.34.1

